ddcommon/src/lib.rs

```diff
@@ -20,6 +25,48 @@ pub mod rate_limiter;
 pub mod tag;
 pub mod tracer_metadata;
 
+#[inline(always)]
+#[track_caller]
+/// Acquires a lock on a `Mutex`, panicking if the lock is poisoned.
+///
+/// This helper function is intended to be used to avoid having to add many
+/// `#[allow(clippy::unwrap_used)]` annotations if there are a lot of usages of `Mutex`.
+///
+/// # Arguments
+///
+/// * `mutex` - A reference to the `Mutex` to lock.
+///
+/// # Returns
+///
+/// A `MutexGuard` that provides access to the locked data.
+///
+/// # Panics
+///
+/// This function will panic if the `Mutex` is poisoned.
+///
+/// # Examples
+///
+/// ```
+/// use ddcommon::lock_or_panic;
+/// use std::sync::{Arc, Mutex};
+///
+/// let data = Arc::new(Mutex::new(5));
+/// let data_clone = Arc::clone(&data);
+///
+/// std::thread::spawn(move || {
+///     let mut num = lock_or_panic(&data_clone);
+///     *num += 1;
+/// })
+/// .join()
+/// .expect("Thread panicked");
+///
+/// assert_eq!(*lock_or_panic(&data), 6);
+/// ```
+pub fn lock_or_panic<T>(mutex: &Mutex<T>) -> MutexGuard<'_, T> {
```

Comment:
Wouldn't that be a good usage of an extension trait? 
```
pub trait MutexExt<T> { fn lock_or_panic(&self) -> MutexGuard<'_, T>; }
impl MutexExt<T> for Mutex<T> { fn lock_or_panic(&self) -> MutexGuard<'_, T> { self.lock.unwrap(); } }
```

It does sort of throw me off that lock_or_panic is prefixed instead of suffixed.

Reply:
Yeah, that's a better idea. Switch to an extension trait.